### ISSUE TITLE
json compile error

### ISSUE THREAD

**Issue by razvancrainea** [OpenSIPS/opensips#1224]
When compiling the json module without `pkg-config`, the version is not properly determined and the compilation fails:
```
make[1]: Entering directory '/root/opensips/modules/json'
/bin/sh: 1: pkg-config: not found
/bin/sh: 1: pkg-config: not found
Compiling json.c
Compiling array_del.c
json.c: In function 'get_object':
json.c:301:4: warning: 'json_object_object_get' is deprecated [-Wdeprecated-declarations]
    cur_obj = json_object_object_get( cur_obj, buff );
    ^~~~~~~
In file included from /usr/include/json-c/linkhash.h:16:0,
                 from /usr/include/json-c/json.h:22,
                 from json.c:48:
/usr/include/json-c/json_object.h:290:56: note: declared here
 THIS_FUNCTION_IS_DEPRECATED(extern struct json_object* json_object_object_get(struct json_object* obj,
                                                        ^
/usr/include/json-c/json_object.h:17:43: note: in definition of macro 'THIS_FUNCTION_IS_DEPRECATED'
 #define THIS_FUNCTION_IS_DEPRECATED(func) func __attribute__ ((deprecated))
                                           ^~~~
In file included from ../../parser/../ip_addr.h:46:0,
                 from ../../parser/msg_parser.h:48,
                 from ../../sr_module.h:48,
                 from json.c:33:
json.c: In function 'pv_set_json':
json.c:595:5: error: 'json_tokener_errors' undeclared (first use in this function)
     json_tokener_errors[(unsigned long)obj]
     ^
../../parser/../dprint.h:340:33: note: in definition of macro 'MY_DPRINT'
      dp_my_pid(), __DP_FUNC, ## args) \
                                 ^~~~
json.c:591:4: note: in expansion of macro 'LM_ERR'
    LM_ERR("Error parsing json: %s\n",
    ^~~~~~
json.c:595:5: note: each undeclared identifier is reported only once for each function it appears in
     json_tokener_errors[(unsigned long)obj]
     ^
../../parser/../dprint.h:340:33: note: in definition of macro 'MY_DPRINT'
      dp_my_pid(), __DP_FUNC, ## args) \
                                 ^~~~
json.c:591:4: note: in expansion of macro 'LM_ERR'
    LM_ERR("Error parsing json: %s\n",
    ^~~~~~
../../Makefile.rules:25: recipe for target 'json.o' failed
make[1]: *** [json.o] Error 1
```
This can be easily replicated on a fresh Debian stretch install, without `pkg-config`

**Comment by adivjoseph** [comment]
I found when enabling json module that include path was out of  date. Since compilers usually look in /usr/include, you need to provide the rest of the path, i.e.:
#include <jsoncpp/json/json.h>
I did this by are still getting errors.
ubuntu LTS 18.05 GCC 5.4.0

**Comment by razvancrainea** [comment]
@adivjoseph actually the `Makefile` should sort this out by adjusting the path to the right include root. Do you have `pkg-config` installed on your system?

PS: also, are you sure that's the correct Ubuntu version? I think Ubuntu 18.04 will be released next April - are you talking about 17.04?

**Comment by razvancrainea** [comment]
I've just tested this and it seems to be all right now. Closing the ticket now.
